win32/devdoc/lazy_init_requirements.md

```diff
@@ -0,0 +1,49 @@
+# lazy_init requirements
+================
+
+## Overview
+
+`lazy_init` is a module that provides the functionality of `lazy_init`, that is, some code called just once for the lifetime of the module. It is threaded: multiple calls to `lazy_init` will allow just one execution of the code that performs the initialization.
+
+## References
+
+
+## Exposed API
+
+```c
+#define LAZY_INIT_NOT_DONE CALL_ONCE_NOT_CALLED /*to only be used in static initialization, rest of initializations need to use interlocked_exchange*/
+
+#define LAZY_INIT_RESULT_VALUES \
+    LAZY_INIT_OK, \
+    LAZY_INIT_ERROR
+
+MU_DEFINE_ENUM(LAZY_INIT_RESULT, LAZY_INIT_RESULT_VALUES)
+
+    typedef int (*LAZY_INIT_FUNCTION)(void* params);
+
+    MOCKABLE_FUNCTION(, LAZY_INIT_RESULT, lazy_init, volatile_atomic int32_t*, lazy, LAZY_INIT_FUNCTION, do_init, void*, init_params);
+
+```
+
+### lazy_init
+```c
+MOCKABLE_FUNCTION(, LAZY_INIT_RESULT, lazy_init, volatile_atomic int32_t*, lazy, LAZY_INIT_FUNCTION, do_init, void*, init_params);

```

Comment:
>init_params [](start = 117, length = 11)

nit: missing "may be NULL" spec/test

Reply:
I thought we don't do that anymore. "if it doesn't look at it... it doesn't validate it". So I fail to see what's the point of having it specified... 

---
In reply to: [466044455](https://github.com/Azure/azure-c-pal/pull/28#discussion_r466044455) [](ancestors = 466044455)